zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/TestStreams.java

```diff
@@ -303,7 +305,7 @@ public void onRecovered(final ReadonlyStreamProcessorContext context) {
                         wrappedFactory, new EngineConfiguration(), new SecurityConfiguration())))
             .streamProcessorMode(streamProcessorMode)
             .maxCommandsInBatch(maxCommandsInBatch)
-            .partitionCommandSender(mock(InterPartitionCommandSender.class))
+            .partitionCommandSender(isReplay ? null : mock(InterPartitionCommandSender.class))
```

Comment:
💭 This is not strictly needed but this is how I discovered that it's not running in `REPLAY`. The command sender is only required for `PROCESSING` and this fails without the change above.

I think it could make sense to keep it so people notice if the processing mode is not respected.